holoviews/operation/element.py

```diff
@@ -583,61 +585,99 @@ def _process(self, element, key=None):
-                        if data_is_datetime[1]:
-                            ys = np.array(num2date(ys))
-                        cp = np.concatenate((xs, ys), axis=1)
-                    if ncp == 0:
-                        exteriors.append(cp)
+        if self.p.filled:
+            empty = np.array([[np.nan, np.nan]])
+            for lower_level, upper_level in zip(levels[:-1], levels[1:]):
+                filled = cont_gen.filled(lower_level, upper_level)
+                # Only have to consider last index 0 as we are using contourpy without chunking
+                if (points := filled[0][0]) is None:
+                    continue
+
+                if any(data_is_datetime[0:2]):
+                    points = points_to_datetime(points)
+
+                offsets = filled[1][0]
+                outer_offsets = filled[2][0]
+                exteriors = []
+                interiors = []
+                for i in range(len(outer_offsets)-1):  # Loop through exterior boundaries
+                    jstart = outer_offsets[i]  # Start boundary index
+                    jend = outer_offsets[i+1]  # End boundary index
+                    if exteriors:
                         exteriors.append(empty)
-                    else:
-                        interior.append(cp)
-                if len(polys):
+                    exteriors.append(points[offsets[jstart]:offsets[jstart + 1]])
+
+                    # Loop over the (jend-jstart-1) interior boundaries,
+                    interior = [points[offsets[j]:offsets[j + 1]] for j in range(jstart+1, jend)]
                     interiors.append(interior)
-            if not exteriors:
-                continue
-            geom = {
-                element.vdims[0].name:
-                num2date(level) if data_is_datetime[2] else level,
-                (xdim, ydim): np.concatenate(exteriors[:-1])
-            }
-            if self.p.filled and interiors:
-                geom['holes'] = interiors
-            paths.append(geom)
+                level = (lower_level + upper_level) / 2
+                geom = {
+                    element.vdims[0].name:
+                    num2date(level) if data_is_datetime[2] else level,
+                    (xdim, ydim): np.concatenate(exteriors) if exteriors else [],
+                }
+                if interiors:
+                    geom['holes'] = interiors
+                paths.append(geom)
+        else:
+            for level in levels:
+                lines = cont_gen.lines(level)
+                # Only have to consider last index 0 as we are using contourpy without chunking
+                if (points := lines[0][0]) is None:
+                    continue
+
+                if any(data_is_datetime[0:2]):
+                    points = points_to_datetime(points)
+
+                offsets = lines[1][0]
+                if len(offsets) > 2:
+                    # Casting offsets to int64 to avoid possible numpy UFuncOutputCastingError
```

Comment:
This is a concern. It isn't needed for `test_operation.py` but is for one of the KDE tests in `test_statsoperations.py`.

Reply:
What was the type before hand?

Reply:
With a clean environment this is now needed to avoid the "UFunc add" error in many tests (`test_operation.py` as well as `test_statsoperations.py`). The complaining line is in numpy `function_base.py` when adding extra values to these `uint32` offsets. The values added could in theory be negative (although they are always positive in practice) and hence it doesn't like the unsigned-ness of `uint32`. `int64` is the next largest signed integer, so it seems a sensible choice.